test/testSimplifier.ts

```diff
@@ -0,0 +1,153 @@
+import * as assert from 'assert';
+import { ModeName } from '../src/mode/mode'
+import { Position } from '../src/motion/position';
+import { ModeHandler } from '../src/mode/modeHandler';
+import { TextEditor } from '../src/textEditor';
+import { assertEqualLines } from './testUtils';
+
+
+interface TestObject {
+    start: string[],
+    keysPressed: string,
+    end: string[],
+    endMode?: ModeName
+}
+
+class TestObjectHelper {
+    startPosition = new Position(0, 0);
+    endPosition = new Position(0, 0);
+    private _isValid = false;
+    
+    constructor(private _testObject: TestObject) {
+        this._parse(_testObject);
+    }
+
+    public get isValid(): boolean {
+        return this._isValid;
+    }
+    
+    private _setStartCursorPosition(lines: string[]) {
+        let result = this._getCursorPosition(lines);
+        this.startPosition = result.position;
+        return result.success;
+    }
+
+    private _setEndCursorPosition(lines: string[]) {
+        let result = this._getCursorPosition(lines);
+        this.endPosition = result.position;
+        return result.success;
+    }
+    
+    private _getCursorPosition(lines: string[]) {
+        let ret = { success: false, position: new Position(0, 0) };
+        lines.forEach((line, lineIdx) => {
```

Comment:
I generally prefer `for` loops over `forEach` etc because they read better and you can more easily `break` out of them if it becomes necessary.